FIAT/serendipity.py

```diff
@@ -122,6 +127,10 @@ def __init__(self, ref_el, degree):
         self.entity_closure_ids = unflattened_entity_closure_ids
         self._degree = degree
         self.flat_el = flat_el
+        if ref_el.get_spatial_dimension() == 2:
+            self.dual = self._get_dual_set_quad()
+        elif ref_el.get_spatial_dimension() == 3:
+            self.dual = self._get_dual_set_hex()
```

Comment:
I think this (= mostly existing code) could be simplified.  There is above a call to superclass `__init__` which would expect the dual set, but `None` is passed in.  Maybe construct the dual set before the superclass call, pass in the dual, and maybe you can also spare all that manual munging after the superclass call.